shared/domains/chelonia/chelonia.js

```diff
@@ -546,22 +546,36 @@ export default (sbp('sbp/selectors/register', {
                     console.info(`[chelonia] Discarding pub event for ${msg.channelID} because it's not in the current subscriptionSet`)
                     return
                   }
-                  const rootState = sbp(this.config.stateSelector)
-                  if (!rootState.contracts[msg.channelID]?.type) {
-                    console.warn(`[chelonia] Discarding pub event for ${msg.channelID} because its contract name could not be determined`)
-                    return
-                  }
                   try {
                     (v: Function)(parseEncryptedOrUnencryptedMessage.call(this, {
                       contractID: msg.channelID,
-                      contractName: rootState.contracts[msg.channelID].type,
                       serializedData: msg.data
                     }))
                   } catch (e) {
                     console.error(`[chelonia] Error processing pub event for ${msg.channelID}`, e)
                   }
                 }
-              : v
+              : k === NOTIFICATION_TYPE.KV
```

Comment:
This is rather difficult to read IMO. Can we use something like a `switch` statement maybe instead? (or a function map?)